File: common-module/src/main/java/ru/kraser/technical_helper/common_module/dto/breakage/BreakageDto.java
package ru.kraser.technical_helper.common_module.dto.breakage;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import ru.kraser.technical_helper.common_module.enums.Priority;
import ru.kraser.technical_helper.common_module.enums.Status;

import java.time.LocalDateTime;

@Builder
public record BreakageDto(
        String id,
        String departmentId,
        String departmentName,      // emp
        String room,                // emp
        String breakageTopic,       // emp
        String breakageText,
        Status status,              // emp
        Priority priority,          // emp
        String executor,            // emp
        String executorAppointedBy,
        String createdBy,           // emp
        @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
        LocalDateTime createdDate,  // emp
        String lastUpdatedBy,
        @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
        LocalDateTime lastUpdatedDate,
        @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
        LocalDateTime deadline
) {
}

File: frontend/src/app/model/breakage/breakage-dto.ts
import { Priority } from "../../enum/priority.enum"
import { Status } from "../../enum/status.enum"

export interface BreakageDto {
    id: String,
    departmentId: String,
    departmentName: String,
    room: String,
    breakageTopic: String,
    breakageText: String,
    status: Status,
    priority: Priority,
    breakageExecutor: String,
    executorAppointedBy: String,
    createdBy: String,
    createdDate: Date,
    lastUpdatedBy: String,
    lastUpdatedDate: Date,
    deadline: Date
}